SmartcarAuth/SmartcarAuth.swift

```diff
@@ -144,19 +143,34 @@ Smartcar Authentication SDK for iOS written in Swift 3.
         let urlComp = URLComponents(url: url, resolvingAgainstBaseURL: false)
 
         guard let query = urlComp?.queryItems else {
-            return completion(AuthorizationError.missingQueryParameters, nil, nil)
+            return completion(AuthorizationError.missingQueryParameters, nil, nil, nil)
         }
 
         let queryState = query.filter({ $0.name == "state"}).first?.value
-
-        if query.filter({ $0.name == "error"}).first?.value != nil {
-            return completion(AuthorizationError.accessDenied, nil, queryState)
+        let vehicle = VehicleInfo()
+        vehicle.vin = query.filter({ $0.name == "vin"}).first?.value ?? ""
+        vehicle.make = query.filter({ $0.name == "make"}).first?.value ?? ""
+        vehicle.model = query.filter({ $0.name == "model"}).first?.value ?? ""
+        if let year = query.filter({ $0.name == "year"}).first?.value {
```

Comment:
I believe you can coerce the type in the if statement here:

```
if let year = query.filter({ $0.name == "year"}).first?.value as? Int {
    vehicle.year = year
}
```

Reply:
changed, thanks!

Reply:
After the changes we did yesterday, it actually wasn't converting so I stopped coercing the type in the if statement.